UI/window-basic-settings.cpp

```diff
@@ -4311,6 +4315,22 @@ void OBSBasicSettings::on_buttonBox_clicked(QAbstractButton *button)
 			return;
 
 		SaveSettings();
+
+#if YOUTUBE_ENABLED
+		std::string service = QT_TO_UTF8(ui->service->currentText());
+		if (IsYouTubeService(service)) {
+			if (!main->GetYouTubeAppDock()) {
+				main->NewYouTubeAppDock();
+			}
+			main->GetYouTubeAppDock()->SettingsUpdated(
+				!IsYouTubeService(service) || stream1Changed);
+		} else {
+			if (main->GetYouTubeAppDock()) {
+				main->GetYouTubeAppDock()->AccountDisconnected();
+			}
+			main->DeleteYouTubeAppDock();
```

Comment:
Should this really be here?
Would it not be better to have this as part of `YouTubeAuth`, similar to the existing chat dock and the integration docks for all other integrations (`TwitchAuth`, `RestreamAuth`)? I think such a solution would generally make this much more clean and mean less changes in places unrelated to the YouTube workflow authentication overall.

Reply:
Hmmmmm. I'll check.

Reply:
Unfotunately, yes. Here.
The reason - YouTubeAuth (and YoutubeApiWrappers) used only when user account connected to the YT. But provided code works with stream id without account connectiong. So, we need to include such calls into OBSBasic.

But I moved code above (and similar) into YouTubeAppDock function, and reduced YT related OBSBasic code to two lines per call.

Reply:
I agree with @gxalpha that this is problematic, is it allows the settings window to directly interfere with the state of the main UI and thus requires the settings window to "know" that such things as YouTube AppDocks exist (it's not a concern of the settings window, so it should not have any concept of this).

IMO this should be handled in the main window when the settings window is closed.

Reply:
For this particular case, it looks better to leave the related code in the settings rather than try to move it to the obsmain. The detailed explanation can be found below:

* dock panel should react to "disconnect account" immediately (this is instant action and can't be undone even if settings are not saved at all) because oauth keys are getting invalidated during disconnect. Similar functionality is already present in [on_disconnectAccount_clicked()](https://github.com/obsproject/obs-studio/blob/master/UI/window-basic-settings-stream.cpp#L784) (805 line, `main->SetBroadcastFlowEnabled(false);` call).
  No separate processing for such a case is present in obsmain, so we made it in a similar way (but without making a function because it is called only once).
* we keep reacting to "connect account" inside settings to not split on/off pair ("connect account"/"disconnect account") of functionality into completely different places. additionally, use case "connect"+"instant disconnect"+"connect"+"instant disconnect"+... is possible and we must process it immediately as described above.
* settings already have youtube-related code and talk with obsmain directly, so we do not introduce anything new in these two blocks(for "connect account" and "disconnect account") of code,
* all discussible functionality (6 and 7 lines of the code in two places) can be enclosed into two new obsmain functions and called as a functions not as a code blocks, but it doesn’t look to be a good idea, because these functions will be called only once and only from the settings.
* if we will try to create a standalone handler for "disconnect account" functionality inside obsmain (like save settings, or like create youtube broadcast), it will increase codebase complexity with only one related code consumer.